lib/index.ts

```diff
@@ -548,6 +550,13 @@ export class UsbbootScanner extends EventEmitter {
 		this.attachedDeviceIds.add(getDeviceId(device));
 
 		const usbbootDevice = this.get(device);
+		let forceSecondstage = false;
+		if (device.deviceDescriptor.iSerialNumber == usbbootDevice?.last_serial) {
```

Comment:
Would this serial number would also be the same if the first stage failed for some reason?